### DISCUSSION TITLE
How to chat with project?

### DISCUSSION THREAD

**Discussion by deividaspetraitis** [yetone/avante.nvim#758] [General]
First of all, thanks for the efforts and time spent implementing and bringing such great project to the community! I was playing with Avante for around and while it worked nice for single scope files fine, I have nocited that there is an option to chat with project context. Now, I just want to clarify that my understanding of this concept is correct before moving on, is my understanding correct that I should be able provide context of all files in the directroy/project and Avante should be able refer to those files in order to generate aswers?

If my understanding is correct, I am having trouble using it, I see there is mentioned `@codebase` keyword here and there, I need use it in the chat, is that right? 

When I try to that when there is no file opened ( after just opening nvim ) it will throw an error:

> prompt: @codebase list all files in project root directory

![image](https://github.com/user-attachments/assets/084b5bc0-2d45-4b50-87f7-b30a4fdb5654)

and hangs due error above:

![image](https://github.com/user-attachments/assets/b587f805-e6ae-4bf3-ac7e-6cc7f531b635)

In all other cases, when I have opened specific file it seems that Avante is still not aware about any other files:

![image](https://github.com/user-attachments/assets/34bf75ce-ba92-4e29-b067-d069ed0b21d6)

Config:

```lua
return {
	"yetone/avante.nvim",
	event = "VeryLazy",
	lazy = false,
	version = false, -- set this if you want to always pull the latest change
	opts = {
		-- add any opts here
	},
	-- if you want to build from source then do `make BUILD_FROM_SOURCE=true`
	build = "make",
	-- build = "powershell -ExecutionPolicy Bypass -File Build.ps1 -BuildFromSource false" -- for windows
	dependencies = {
		"nvim-treesitter/nvim-treesitter",
		"stevearc/dressing.nvim",
		"nvim-lua/plenary.nvim",
		"MunifTanjim/nui.nvim",
		--- The below dependencies are optional,
		"nvim-tree/nvim-web-devicons", -- or echasnovski/mini.icons
		"zbirenbaum/copilot.lua", -- for providers='copilot'
		{
			-- support for image pasting
			"HakonHarnes/img-clip.nvim",
			event = "VeryLazy",
			opts = {
				-- recommended settings
				default = {
					embed_image_as_base64 = false,
					prompt_for_file_name = false,
					drag_and_drop = {
						insert_mode = true,
					},
					-- required for Windows users
					use_absolute_path = true,
				},
			},
		},
		{
			-- Make sure to set this up properly if you have lazy=true
			'MeanderingProgrammer/render-markdown.nvim',
			opts = {
				file_types = { "markdown", "Avante" },
			},
			ft = { "markdown", "Avante" },
		},
	},

	config = function()
		require('avante').setup {
			provider = "claude",         -- Recommend using Claude
			auto_suggestions_provider = "claude", -- Since auto-suggestions are a high-frequency operation and therefore expensive, it is recommended to specify an inexpensive provider or even a free provider: copilot
			claude = {
				endpoint = "https://api.anthropic.com",
				model = "claude-3-5-sonnet-20240620",
				temperature = 0,
				max_tokens = 4096,
			},
			behaviour = {
				auto_suggestions = false, -- Experimental stage
			},
			mappings = {
				submit = {
					normal = "<CR>",
					insert = "<C-y>",
				},
				suggestion = {
					accept = "<C-l>",
					next = "<C-n>",
					prev = "<C-p>",
					dismiss = "<C-q>",
				},
			},
		}
	end
}
```

Note1: I am testing with Go project
Note2: I have tried both `AvanteChat` and `AvanteAsk` commands

**Comment by yetone** [comment]
‌‌‌‌There is currently a repomap, but the effect is not good. Next, I will consider introducing CoT to better complete multi-file programming tasks.

**Reply by deividaspetraitis** [reply]
Got it, thanks for prompt reply! 🙏
